test/extensions/clusters/aggregate/cluster_test.cc

```diff
@@ -0,0 +1,303 @@
+    EXPECT_CALL(cm_, get(Eq("tertiary"))).WillRepeatedly(Return(nullptr));
+    ON_CALL(primary_, prioritySet()).WillByDefault(ReturnRef(primary_ps_));
+    ON_CALL(secondary_, prioritySet()).WillByDefault(ReturnRef(secondary_ps_));
+    ON_CALL(aggregate_cluster_, loadBalancer()).WillByDefault(ReturnRef(*lb_));
+
+    setupPrioritySet();
+  }
+
+  Stats::IsolatedStoreImpl stats_store_;
+  Ssl::MockContextManager ssl_context_manager_;
+  NiceMock<Upstream::MockClusterManager> cm_;
+  NiceMock<Runtime::MockRandomGenerator> random_;
+  NiceMock<ThreadLocal::MockInstance> tls_;
+  NiceMock<Runtime::MockLoader> runtime_;
+  NiceMock<Event::MockDispatcher> dispatcher_;
+  NiceMock<LocalInfo::MockLocalInfo> local_info_;
+  NiceMock<Server::MockAdmin> admin_;
+  Singleton::ManagerImpl singleton_manager_{Thread::threadFactoryForTest()};
+  NiceMock<ProtobufMessage::MockValidationVisitor> validation_visitor_;
+  Api::ApiPtr api_{Api::createApiForTest(stats_store_)};
+  std::shared_ptr<Cluster> cluster_;
+  Upstream::ThreadAwareLoadBalancerPtr thread_aware_lb_;
+  Upstream::LoadBalancerFactorySharedPtr lb_factory_;
+  Upstream::LoadBalancerPtr lb_;
+  Upstream::ClusterStats stats_;
+  std::shared_ptr<Upstream::MockClusterInfo> info_{new NiceMock<Upstream::MockClusterInfo>()};
+  NiceMock<Upstream::MockThreadLocalCluster> aggregate_cluster_, primary_, secondary_;
+  Upstream::PrioritySetImpl primary_ps_, secondary_ps_;
+  NiceMock<Upstream::MockLoadBalancer> primary_load_balancer_, secondary_load_balancer_;
+
+  const std::string default_yaml_config_ = R"EOF(
+    name: aggregate_cluster
+    connect_timeout: 0.25s
+    lb_policy: CLUSTER_PROVIDED
+    cluster_type:
+      name: envoy.clusters.aggregate
+      typed_config:
+        "@type": type.googleapis.com/envoy.config.cluster.aggregate.ClusterConfig
+        clusters:
+        - primary
+        - secondary
+)EOF";
+}; // namespace Aggregate
+
+TEST_F(AggregateClusterTest, LoadBalancerTest) {
+  initialize(default_yaml_config_);
+  // Health value:
+  // Cluster 1:
+  //     Priority 0: 33.3%
+  //     Priority 1: 33.3%
+  // Cluster 2:
+  //     Priority 0: 33.3%
+  //     Priority 1: 33.3%
+  Upstream::HostSharedPtr host = Upstream::makeTestHost(info_, "tcp://127.0.0.1:80");
+  EXPECT_CALL(primary_, loadBalancer()).WillRepeatedly(ReturnRef(primary_load_balancer_));
+  EXPECT_CALL(primary_load_balancer_, chooseHost(_)).WillRepeatedly(Return(host));
+  EXPECT_CALL(secondary_, loadBalancer()).WillRepeatedly(ReturnRef(secondary_load_balancer_));
+  EXPECT_CALL(secondary_load_balancer_, chooseHost(_)).WillRepeatedly(Return(nullptr));
+
+  for (int i = 0; i <= 65; ++i) {
+    EXPECT_CALL(random_, random()).WillOnce(Return(i));
+    Upstream::HostConstSharedPtr target = lb_->chooseHost(nullptr);
+    EXPECT_EQ(host.get(), target.get());
+  }
+
+  EXPECT_CALL(primary_, loadBalancer()).WillRepeatedly(ReturnRef(primary_load_balancer_));
```

Comment:
since you're always returning primary_load_balancer_ for primary etc., i would just do the expectations at the start and rely on `WillRepeatedly` to avoid having to repeat them all the time